InnoClinic.ProfilesAPI/Infrastructure/Repositories/DoctorRepository.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Domain.Entities;
+using Domain.Interfaces;
+using Infrastructure.Context;
+using Microsoft.EntityFrameworkCore;
+
+namespace Infrastructure.Repositories;
+
+public class DoctorRepository(ApplicationDbContext context) : IDoctorRepository
+{
+    public async Task<Doctor> CreateDoctorAsync(Doctor doctor, CancellationToken cancellationToken = default)
+    {
+        await context.Doctors.AddAsync(doctor, cancellationToken);
+        await context.SaveChangesAsync(cancellationToken);
+        return doctor;
+    }
+
+    public async Task DeleteDoctorAsync(Doctor doctor, CancellationToken cancellationToken = default)
+    {
+        context.Doctors.Remove(doctor);
+        await context.SaveChangesAsync(cancellationToken);
+    }
+
+    public async Task<Doctor?> GetDoctorByIdAsync(Guid id, CancellationToken cancellationToken = default) =>
+        await context.Doctors.FindAsync(id, cancellationToken);
```

Comment:
As you know some EF theory, I recommend you to add method parameter to use or don't use AsNoTraking. For GET requests better to skip tracking to improve performance, but it's needed to update something.